opencog/nlp/sureal/SuRealPMCB.cc

```diff
@@ -73,32 +80,60 @@ bool SuRealPMCB::variable_match(Handle &hPat, Handle &hSoln)
     std::string sPat = _as->getName(hPat);
     std::string sPatWord = sPat.substr(0, sPat.find_first_of('@'));
     std::string sSoln = _as->getName(hSoln);
-    std::string sSolnWord = sSoln.substr(0, sSoln.find_first_of('@'));
 
     // get the WordNode associated with the word (extracted from "word@1234" convention)
     Handle hPatWordNode = _as->getHandle(WORD_NODE, sPatWord);
-    Handle hSolnWordNode = _as->getHandle(WORD_NODE, sSolnWord);
 
-    // no WordNode? reject!
-    if (hSolnWordNode == Handle::UNDEFINED)
+    // get the corresponding WordInstanceNode for hSoln
+    Handle hSolnWordInst = _as->getHandle(WORD_INSTANCE_NODE, sSoln);
+
+    // no WordInstanceNode? reject!
+    if (hSolnWordInst == Handle::UNDEFINED)
         return false;
 
-    // get the neighbor of each WordNode within LgWordCset
-    HandleSeq qPatSet =_as->getNeighbors(hPatWordNode, false, true, LG_WORD_CSET, false);
-    HandleSeq qSolnSet = _as->getNeighbors(hSolnWordNode, false, true, LG_WORD_CSET, false);
+    // get the source connectors for the solution
+    std::string scmCode = "(word-inst-get-source-conn " + SchemeSmob::to_string(hSolnWordInst) + ")";
```

Comment:
to avoid having to use eval_q here, you could just say
```
std::string scmCode = "(ListLink (word-inst-get-source-conn " ...
 HandleSeq qTargetConns = m_eval->eval_h(scmCode)->getOutgoingSet();
```
Its a little more verbose ...